feat(Wiki.gg): add presence

DarkVillager opened the pull request:
## Description 
<!-- A clear and detailed description of the changes, referencing issues if applicable -->
Resolves #6257
Added the presence.
## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [x] I bumped the version of the presence in accordance with [Semver](https://devhints.io/semver)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<!-- 
    Screenshots of the presence settings (if applicable) and at least TWO screenshots of the presence displaying correctly
    Including these screenshots will assist the reviewing processes thus speeding up the process of the pull request being merged
-->
<br>
<details>
<summary> Terraria</summary>
<img src="https://user-images.githubusercontent.com/42322979/169804182-38b0ae09-6140-484c-a851-73c8b816cb68.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804184-b63f0f7f-abd7-4812-9e07-1aeef9f06b72.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804186-3cc16a83-29bb-4e22-8497-8b8b3f916d65.jpg">

</details>

<details>
<summary> Coromon</summary>
<img src="https://user-images.githubusercontent.com/42322979/169804484-4b7c81d3-1a3f-4021-add9-584f31517f49.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804489-cb7ea370-cc53-40f7-862e-73613e0cf04c.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804490-68e5e634-a498-44de-bb93-a9793f56dc45.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804491-d5252a07-66d4-4c84-ade8-6c1327277f0d.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804964-b0285027-f7d1-4f6d-aa49-b8710fb1d2f9.png">

</details>

<details>
<summary> Darkdeity</summary>
<img src="https://user-images.githubusercontent.com/42322979/169804506-07cc64f0-29a4-417b-8353-a268bc640f7d.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804513-656d33f9-8442-46f4-81d7-c1ea626c9b87.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804516-b0e3c145-e7f8-4c83-9902-12b535c1fd51.jpg">

</details>

<details>
<summary> Wiki.gg</summary>
<img src="https://user-images.githubusercontent.com/42322979/169804164-6270a6fa-34c2-4c79-b5d7-a864ed61c666.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804170-ceaaaf29-0316-4e31-8982-034bdafec9e9.jpg">
<img src="https://user-images.githubusercontent.com/42322979/169804171-1213beba-8a4a-442c-af93-a84c40088175.jpg">

</details>

</details>

DarkVillager committed:
feat(Wiki.gg): add presence

style: fix formatting issues

style: fix formatting issues

style: fix formatting issues

DarkVillager committed:
fix(Wiki.gg): lint

EncryptedDev commented on websites/W/Wiki.gg/presence.ts:
```diff
@@ -0,0 +1,80 @@
+		clientId: "978186598669758504"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+let title: string, mTitle: string, search: NodeListOf<HTMLInputElement>;
```
No

EncryptedDev commented on websites/W/Wiki.gg/presence.ts:
```diff
@@ -0,0 +1,80 @@
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp
+		},
+		path = document.location.pathname,
```
```suggestion
		{ pathname } = document.location,
```

EncryptedDev commented on websites/W/Wiki.gg/presence.ts:
```diff
@@ -0,0 +1,80 @@
+					.textContent.replace(":", " ");
+				presenceData.buttons = [
+					{
+						label: "View Wiki Page",
```
```suggestion
						label: "View Page",
```

EncryptedDev commented on websites/W/Wiki.gg/presence.ts:
```diff
@@ -0,0 +1,80 @@
+				)
+					presenceData.state = search[1].value;
+				title = `Searching ${title}'s wiki for:`;
+				presenceData.buttons = [
+					{
+						label: "View Search Result",
+						url: document.location.href
+					}
+				];
```
You don't need a button here

EncryptedDev commented on websites/W/Wiki.gg/presence.ts:
```diff
@@ -0,0 +1,80 @@
+					"#mw-content-text > div.mw-parser-output > div > div:nth-child(1) > div > div:nth-child(1)"
+				)
+			)
+				presenceData.state = "Viewing Homepage";
```
```suggestion
				presenceData.state = "Viewing home page";
```

DarkVillager replied:
![image](https://user-images.githubusercontent.com/42322979/171164785-8c273e14-e0bb-4033-a05c-d7c645b2220a.png)

EncryptedDev replied:
I'm running of our strings
https://github.com/PreMiD/Localization/blob/46616f565a2a05592e8ae58cab697da9516e06eb/src/Presence/general.json#L162-L164

EncryptedDev replied:
Resolved

DarkVillager committed:
fix(Wiki.gg): lint issues and pathname

fix(Wiki.gg): pathname and removed search button

fix(Wiki.gg): changed let to const

EncryptedDev commented on websites/W/Wiki.gg/presence.ts:
```diff
@@ -0,0 +1,76 @@
+				presenceData.state = "Viewing Homepage";
+			presenceData.details = title;
+		}
+	} else presenceData.details = "Browsing...";
```
```suggestion
	} else presenceData.details = "Browsing";
```

EncryptedDev commented on websites/W/Wiki.gg/dist/metadata.json:
```diff
@@ -0,0 +1,41 @@
+	"service": "Wiki.gg",
+	"description": {
+		"en": "Platform operated by creators of gaming experiences. Flexible platform that adjusts to the needs of creators.",
+		"nl": "Platform beheerd door makers van spelervaringen. Flexibel platform dat zich aanpast aan de behoeften van de makers."
```
Suggestion from @QkeleQ10 
```suggestion
		"nl": "Wij brengen een gemeenschap van ontwikkelaars en editors samen om de best mogelijke ervaring voor gamers te creëren."
```

DarkVillager committed:
fix(Wiki.gg): capitals and metadata

EncryptedDev commented on websites/W/Wiki.gg/presence.ts:
```diff
@@ -0,0 +1,76 @@
+		clientId: "978186598669758504"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+let title: string, mTitle: string;
```
What is `mTitle`?

EncryptedDev committed:
fix(Wiki.gg): clearer variable name and wording

fix: inconsistent wording

EncryptedDev commented:
Heading towards being inactive, please answer the review above

ImRodry commented:
Rebased to apply the latest changes in the prettier configuration. If you need help with your local branch, please ping me on the #presence-dev channel on Discord.

ImRodry commented:
Please fix the issues with ESLint

DarkVillager committed:
Update presence.ts

EncryptedDev reviewed (approved):
LGTM
